lib/improver/utilities/cube_manipulation_new.py

```diff
@@ -38,100 +38,44 @@
-    auxiliary coordinates, where these auxiliary coordinates will be
-    associated with the master coordinate.
-
-    For example, forecast_reference_time and forecast_period can be converted
-    from scalar coordinates to auxiliary coordinates, and associated with time.
+    Function to remove attributes that do not match between all cubes on the
+    list.  Cubes are modified in place.
 
     Args:
-        cube (Iris cube):
-            Cube requiring addition of the specified coordinates as auxiliary
-            coordinates.
-        master_coord (String):
-            Coordinate that the other coordinates will be associated with.
-        coordinates (None or List):
-            List of coordinates to be associated with the master_coord.
-
-    Returns:
-        forecast_data (Iris cube):
-            Cube where the the requested coordinates have been added to the
-            cube as auxiliary coordinates and associated with the desired
-            master coordinate.
-
-    Raises:
-        ValueError: If the master coordinate is not present on the cube.
-
-    """
-    if coordinates is None:
-        coordinates = []
-
-    if not cube.coords(master_coord):
-        msg = (
-            "The master coordinate for associating other "
-            "coordinates with is not present: "
-            "master_coord: {}, other coordinates: {}".format(
-                master_coord, coordinates))
-        raise ValueError(msg)
-
-    # If the master_coord is not a dimension coordinate, then the other
-    # coordinates cannot be associated with it.
-    if len(cube.coords(master_coord, dim_coords=True)) > 0:
-        for coord in coordinates:
-            if cube.coords(coord):
-                temp_coord = cube.coord(coord)
-                cube.remove_coord(coord)
-                temp_aux_coord = (
-                    build_coordinate(temp_coord.points,
-                                     bounds=temp_coord.bounds,
-                                     coord_type=AuxCoord,
-                                     template_coord=temp_coord))
-                coord_names = [
-                    coord.standard_name for coord in cube.dim_coords]
-                cube.add_aux_coord(
-                    temp_aux_coord,
-                    data_dims=coord_names.index(master_coord))
-    return cube
-
-
-def _slice_over_coordinate(cubes, coord_to_slice_over):
-    """
-    Function slice over the requested coordinate,
-    promote the sliced coordinate into a dimension coordinate
-    to help concatenation.
-
-    Args:
-        cubes (Iris cubelist or Iris cube):
-            Cubes to be concatenated.
-        coords_to_slice_over (List):
-            Coordinates to be sliced over.
+        cubes (iris.cube.CubeList):
+            List of cubes to check the attributes and revise.
 
-    Returns:
-        Iris CubeList
-            CubeList containing sliced cubes.
+    Kwargs:
+        unmatched (list or None):
```

Comment:
Is there a use case for `unmatched` being a keyword argument, rather than just being computed by the `compare_attributes` function?

Reply:
I think my initial thought was that I wanted to be able to ignore a mis-matched "model_id" attribute, or equivalent, when merging.  But given how the code has come out I'm not sure this still applies.  I'll have a proper think and come back to this.

Reply:
OK I've had a look at the more complete version and this argument isn't actually needed.  Given the complexity for the user, I'll remove the `unmatched` option.